src/index.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Components using the react-intl module require access to the intl context.
+ * This is not available when mounting single components in Enzyme.
+ * These helper functions aim to address that and wrap a valid,
+ * English-locale intl context around them.
+ */
+import { cloneElement } from 'react';
+import { IntlProvider, intlShape } from 'react-intl';
+import { mount, shallow, render } from 'enzyme';
+
+export const mockIntl = {
+  defaultLocale: 'en',
+  formatDate: () => 'mock date',
+  formatHTMLMessage: ({ id }) => id,
+  formatMessage: ({ id }) => id,
+  formatNumber: value => `${value}`,
+  formatPlural: value => `${value}`,
+  formatTime: value => `${value}`,
+  formatRelative: value => `${value}`,
+  now: () => 0,
+};
+
+const messageProxy = new Proxy({}, { get: (_, property) => property });
+
+// Create the IntlProvider to retrieve context for wrapping around.
+const intlProvider = new IntlProvider({ locale: 'en', messages: messageProxy }, {});
+const { intl } = intlProvider.getChildContext();
+
+const injectIntl = node => cloneElement(node, { intl });
+
+export const shallowContext = { context: { intl } };
+export const mountContext = { context: { intl }, childContextTypes: { intl: intlShape } };
```

Comment:
I have some concerns around exporting context for usage in this format. My preference is to provide the withIntl functions and not export `shallowContext` or `mountContext`. 

The idea behind this preference is that the withIntl functions offer an abstraction layer of the react-intl context. If that context implementation were to change, my hope would be that we can account for it within the withIntl functions with minimal changes needed by consumers that use these functions. 

By exporting `shallowContext` and `mountContext`, if/when the context changes in react-intl, it seems like it would be more difficult to provide a clean upgrade path for consumers that use those exports.

I think of this as being similar to using injectIntl (which is an abstraction layer of the react-intl context) vs interfacing directly with react-intl context to get the intl object. 

We recommend using the abstraction layer with interfacing with react-intl's Intl object when creating components, I think we should recommend interfacing with an abstraction layer as well when testing components.

Reply:
I think that the new Context API is going to cause a large shift in how we unit test context-consuming components in general. Any components using the new APIs won't be able to directly receive values through Enzyme's shallow/mount functions. Enzyme has an open PR to add some support for the new Context API, but those changes would necessitate that we shallow render the IntlProvider and subsequently dive()-ing until we get to the component we want to test.

Intl isn't a one-off. We are going to run into this more and more as we add Context-powered components, and our consumers that are consuming multiple contexts are going to feel the pain very quickly. We need to come up with a good overall strategy and hopefully provide some utilities to reduce friction in our tests. `shallowWithIntl`/`mountWithIntl` are first class solutions for a single Context implementation; we need something akin to `shallowWithContexts` to handle things more generally. 

To that end, I'm not sure exporting these would be a huge deal in the meantime if they're providing value today. But these helpers will be short-lived with react-intl's transition to the new API and the deprecation of the old context APIs by React itself. I think anything we need to do to update in the future will be so drastic as to render passivity here a moot point.